core/src/main/scala/monocle/function/Snoc1.scala

```diff
@@ -20,7 +20,12 @@ trait Snoc1[S, I, L] extends Serializable {
   def last: Lens[S, L] = snoc1 composeLens second
 }
 
-object Snoc1 extends Snoc1Functions
+object Snoc1 extends Snoc1Functions {
+  def fromIso[S, A, I, L](iso: Iso[S, A])(implicit ev: Snoc1[A, I, L]): Snoc1[S, I, L] = new Snoc1[S, I, L] {
+    override def snoc1: Iso[S, (I, L)] =
+      iso composeIso ev.snoc1
```

Comment:
Add documentation?
This won't compose from the other side like in Snoc because we have freedom in the tail; for example, NEL[A] = (A,List[A])